textselect.cpp

```diff
@@ -140,6 +140,39 @@ TextSelect::Selection TextSelect::getSelection() const {
     return { startX, startY, endX, endY };
 }
 
+//
+// Taken from imgui_draw.cpp
+//
+// Trim trailing space and find beginning of next line
+static inline const char* CalcWordWrapNextLineStartA(const char* text, const char* text_end)
+{
+    while (text < text_end && ImCharIsBlankA(*text))
+        text++;
+    if (*text == '\n')
+        text++;
+    return text;
+}
+
+// Split `text` that does not fit in `wrapWidth` into multiple lines.
+static ImVector<std::string_view> wrapText(std::string_view text, float wrapWidth, ImFont *font) {
+    ImVector<std::string_view> result;
+    const char *textEnd = text.data() + text.size();
+    const char *wrappedLineStart = text.data();
+    const char *wrappedLineEnd = text.data();
+    while (wrappedLineEnd != textEnd) {
+        wrappedLineStart = wrappedLineEnd;
+        wrappedLineEnd = font->CalcWordWrapPositionA(1, wrappedLineStart, textEnd, wrapWidth);
+
+        if (wrappedLineEnd - wrappedLineStart != 0) {
+            result.push_back(std::string_view(wrappedLineStart, wrappedLineEnd));
```

Comment:
We can't use the (start, end) constructor since it was introduced in C++20 and this library uses C++17. Also, using brace-initialization and an explicit `std::size_t` cast matches the rest of the code.

```suggestion
            result.push_back({ wrappedLineStart, static_cast<std::size_t>(wrappedLineEnd - wrappedLineStart) });
```